### ISSUE TITLE
parent-module not running on bun

### ISSUE THREAD

**Issue by noraj** [oven-sh/bun#4742]
### What version of Bun is running?

1.0.0+822a00c4d508b54f650933a73ca5f4a3af9a7983

### What platform is your computer?

Linux 6.4.12-arch1-1 x86_64 unknown

### What steps can reproduce the bug?

- bun x create-docusaurus@latest my-website classic
- cd my-website
- bun install
- bun run docusaurus

### What is the expected behavior?

docusaurus running like on nodejs

### What do you see instead?

```
➜ bun run docusaurus
$ docusaurus
 9 |    }
10 | 
11 |    let seenVal = false;
12 | 
13 |    // Skip the first stack as it's this function
14 |    stacks.shift();
    ^
TypeError: stacks.shift is not a function. (In 'stacks.shift()', 'stacks.shift' is undefined)
      at /tmp/my-website/node_modules/parent-module/index.js:14:1
      at /tmp/my-website/node_modules/import-fresh/index.js:11:20
      at /tmp/my-website/node_modules/@docusaurus/core/lib/server/config.js:36:10
      at processTicksAndRejections (:1:2602)
error: script "docusaurus" exited with code 1 (SIGHUP)
```

### Additional information

> If you run into any bugs with a particular package, please [open an issue](https://bun.sh/issues). Opening issues for compatibility bugs helps us prioritize what to work on next.

Asking for bun support on upstream https://github.com/facebook/docusaurus/issues/9294

**Comment by Josh-Cena** [comment]
Note that the stack ultimately points to `parent-module`. The code looks like:

```js
const callsites = require('callsites');

module.exports = filePath => {
	const stacks = callsites();
// ...
```

`callsites` looks like this:

```js
const callsites = () => {
	const _prepareStackTrace = Error.prepareStackTrace;
	Error.prepareStackTrace = (_, stack) => stack;
	const stack = new Error().stack.slice(1);
	Error.prepareStackTrace = _prepareStackTrace;
	return stack;
};
```

Therefore, it's probably because JSC doesn't have `Error.prepareStackTrace`. Bun would probably need to polyfill these V8 APIs.

The issue title is misleading because it has nothing to do with Docusaurus.

**Comment by noraj** [comment]
This issue may be overlapping with #3426 and #4324

**Comment by Jarred-Sumner** [comment]
Fixed in https://github.com/oven-sh/bun/pull/5802

This will be released in Bun v1.0.3. 

If you don't want to wait until then, an hour after this comment was written you should be able to upgrade to the canary build of Bun:

```bash
bun upgrade --canary
```
